src/kernels/types.ts

```diff
@@ -237,6 +237,10 @@ export interface IKernelProvider extends IAsyncDisposable {
      * WARNING: If called with different options for same Notebook, old kernel associated with the Uri will be disposed.
      */
     getOrCreate(uri: Uri, options: KernelOptions): IKernel;
+    /**
```

Comment:
Feels like this would make more sense to be on the IKernel itself. Why does the provider even have to know about the wrappers?

Reply:
Keeping it here, as the kernelProvider is responsible for determining whether a new controller needs to be created or not when using `getorCreate` and providing the controller, metedata information & the like.